marlowe/test/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -419,6 +419,64 @@ choiceNotInBounds bounds =
     suchThat arbitrary $ \chosenNum -> not $ any (inBound chosenNum) bounds
 
 
+-- | Generate relevant Input content for a given input action
+interestingInput :: Bool -> Action -> [InputContent]
+interestingInput validity (Choice choiceId) bounds =
+  map (IChoice choiceId) (interestingChoiceNum bounds)
+  ++ map (\choiceId -> IChoice choiceId bounds) (interestingChoiceId choiceId)
+interestingInput validity (Choice choiceId) bounds = IChoice choiceId <$> interestingChoiceNum' validity bounds
+
+interestingChoiceId :: ChoiceId -> [ChoiceId]
+interestingChoiceId _ = error "not implemented yet"
+
+interestingParties :: Party -> [Party]
+interestingParties (Role "Alice") = [Role "Alice", Role "Alic", Role "Alices"]
+
+interestingChoiceNum :: [Bound] -> [ChoiceNum]
+interestingChoiceNum bounds = concatMap interestingChoiceNum' bounds
+
+interestingChoiceNum' :: Bool -> Bound -> [ChoiceNum]
+interestingChoiceNum' True (Bound lower upper)
+  | lower > upper = Nothing
```

Comment:
@bwbush Do we want to raise an error in this case? Not sure how to handle this.

Reply:
No, since the error would be caught and dealt with elsewhere. In this case nothing is interesting, so it should return `[]`.